projects/observability/src/shared/components/gauge/gauge-chart.component.ts

```diff
@@ -0,0 +1,99 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import { TypedSimpleChanges } from '@hypertrace/common';
+import { maxBy } from 'lodash-es';
+import { HistogramBarData } from '../histogram/histogram-chart.component';
+
+@Component({
+  selector: 'ht-gauge-chart',
+  styleUrls: ['./gauge-chart.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="gauge-chart">
+      <ng-container *ngFor="let item of this.itemOptions">
+        <div class="border-top"></div>
+        <div
+          class="label"
+          [htcTooltip]="item.label"
+          (click)="this.onItemClick(item)"
+          [ngClass]="{ clickable: this.labelClickable }"
+          [ngStyle]="{ color: item.color }"
+        >
+          {{ item.label }}
+        </div>
+        <div class="progress">
+          <div class="progress-value" [ngStyle]="{ width: item.width, backgroundColor: item.color }"></div>
+        </div>
+        <div class="value">
+          <span>{{ item.value | htcDisplayNumber }}</span>
+        </div>
+      </ng-container>
+    </div>
+  `
+})
+export class GaugeChartComponent<T extends GaugeData = GaugeData> implements OnChanges {
+  @Input()
+  public data: T[] = [];
+
+  @Input()
+  public labelClickable: boolean = false;
```

Comment:
If we're refactoring this anyway, I think it'd be a good opportunity to clean up the api. We take in `data`, use `labelClickable` to determine if it can be clicked and `itemClick` to ouput that click - all three are really referring to the same unit.

Reply:
What do you mean by clean the api though?

Reply:
data, label and item all refer to the same thing. So perhaps `data` ->`items` `labelClickable` -> `itemClickable` for example

Reply:
ah, just renames. makes sense